app/services/katello/pulp/repository.rb

```diff
@@ -120,6 +120,56 @@ def mirror_importer_connection_options
       def proxy_host_importer_value
         root.ignore_global_proxy ? "" : nil
       end
+
+      def refresh
+        repository_details = backend_data
+        update_or_associate_importer(repository_details)
+        update_or_associate_distributors(repository_details)
+        remove_unnecessary_distributors(repository_details)
+      end
+
+      def update_or_associate_importer(repository_details)
+        existing_importers = repository_details["importers"]
+        importer = generate_importer
+        found = existing_importers.find { |i| i['importer_type_id'] == importer.id }
+
+        if found
+          ssl_ca_cert = importer.config.delete('ssl_ca_cert')
+          ssl_client_cert = importer.config.delete('ssl_client_cert')
+          ssl_client_key = importer.config.delete('ssl_client_key')
+          importer.config['basic_auth_username'] = nil if importer.config['basic_auth_username'].blank?
+          importer.config['basic_auth_password'] = nil if importer.config['basic_auth_password'].blank?
+          smart_proxy.pulp_api.resources.repository.update_importer(repo.pulp_id, found['id'], :ssl_client_cert => ssl_client_cert,
+                                                    :ssl_client_key => ssl_client_key, :ssl_ca_cert => ssl_ca_cert)
+          smart_proxy.pulp_api.resources.repository.update_importer(repo.pulp_id, found['id'], importer.config)
```

Comment:
why is update_importer being called twice?  shouldn't importer.config handle it?

Reply:
Calling update_importer twice is probably the workaround in place for this issue https://pulp.plan.io/issues/2727 as per comment on issue and in code.
https://github.com/Katello/katello/blob/82d9aceca07aa218f6deefff7df74b0ee4b87ff6/app/lib/actions/pulp/repository/update_importer.rb#L13

Reply:
can you add that comment here?  so its more obvious why this is happening.